src/main/java/htsjdk/samtools/BAMIndexMetaData.java

```diff
@@ -147,23 +148,25 @@ void recordMetaData(final SAMRecord rec) {
     }
 
     /**
-     * @param slice
+     * Record metadata for a given CRAM slice
+     *
+     * @param sliceMetadata Information about a CRAM slice.
      */
-    void recordMetaData(Slice slice) {
+    void recordMetaData(final SliceBAIMetadata sliceMetadata) {
 
-        final int alignmentStart = slice.alignmentStart;
+        final int alignmentStart = sliceMetadata.getAlignmentStart();
         if (alignmentStart == SAMRecord.NO_ALIGNMENT_START) {
-            noCoordinateRecords+=slice.nofRecords;
+            noCoordinateRecords += sliceMetadata.getRecordCount();
             return;
         }
 
-        final long start = slice.offset;
-        final long end = slice.offset + 0;

```

Comment:
This made no sense to me - please let me know if there was a reason for this line.

Reply:
I'll revert and point to the open ticket #401